File: src/top150/n28_find_the_index_of_the_first_occurrence_in_a_string/solution.go
package main

import "fmt"

func strStr(haystack, needle string) int {
	if len(needle) > len(haystack) {
		return -1
	}

	for i := 0; i <= len(haystack)-len(needle); i++ {
		if haystack[i:i+len(needle)] == needle {
			return i
		}
	}

	return -1
}

func main() {
	haystack := "hello"
	needle := "ll"
	result := strStr(haystack, needle)
	fmt.Println(result)
}

File: src/top150/n28_find_the_index_of_the_first_occurrence_in_a_string/Solution.java
package top150.n28_find_the_index_of_the_first_occurrence_in_a_string;

public class Solution {

    public int strStr(String haystack, String needle) {
        if (needle.length() > haystack.length())
            return -1;

        for (int i = 0; i < haystack.length() - needle.length(); i++) {
            if (haystack.substring(i).startsWith(needle)) {
                return i;
            }
        }

        return -1;
    }

    public static void main(String[] args) {
        var s = new Solution();
        System.out.println(s.strStr("sadbutsad", "sad"));
    }
}
